internal/rukpak/util/util.go

```diff
@@ -0,0 +1,300 @@
+package util
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	rukpakv1alpha2 "github.com/operator-framework/rukpak/api/v1alpha2"
+)
+
+var (
+	// ErrMaxGeneratedLimit is the error returned by the BundleDeployment controller
+	// when the configured maximum number of Bundles that match a label selector
+	// has been reached.
+	ErrMaxGeneratedLimit = errors.New("reached the maximum generated Bundle limit")
+)
+
+func BundleDeploymentProvisionerFilter(provisionerClassName string) predicate.Predicate {
```

Comment:
We are not using Provisioners here, so I think all the methods in this file can be removed and would not be useful.